### ISSUE TITLE
[BERT] Reproduce BERT on SWAG 

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#599]
Anyone wants to help reproduce the result reported by [BERT](https://arxiv.org/pdf/1810.04805.pdf ) on the grounded commonsense inference task on SWAG?

**Comment by eric-haibin-lin** [comment]
Suggestion from the author regarding implementation
https://github.com/google-research/bert/issues/38

**Comment by omerarshad** [comment]
still not able to get 81.6 on base model, all i can get is 80.16

**Comment by eric-haibin-lin** [comment]
hi @omerarshad what is your setup and what hyper-parameters did you try?

**Comment by omerarshad** [comment]
learning rate 2e-5
batch size 16
model bert-base-uncased
gradient accumulation 4

**Comment by eric-haibin-lin** [comment]
Hi @omerarshad sorry for the late reply. I noticed that in the paper the author used the following hyper-parameters:
```
We fine-tune the model for 3 epochs with a learning rate of 2e-5 and a batch size of 16.
```
In your case, since you used batch_size = 16 and grad_accumulation=4, your effective batch size is 64. 
What about setting epoch=3, batchsize=16, accumulate=1 and trying again?

Nonetheless 80.16 is a good starting point and is much better than the ELMo baseline. Hopefully we can get your work into gluonnlp :)

**Comment by eric-haibin-lin** [comment]
@omerarshad any luck?

**Comment by omerarshad** [comment]
80.69 was last i achieved week ago.

**Comment by szha** [comment]
@omerarshad thanks! what were the hyperparameters?

**Comment by eric-haibin-lin** [comment]
@omerarshad did you check average length of the corpus and the max length used for finetuning? I think the default in the script is 128

**Comment by mingtop** [comment]
I have got 80.96 on bert-base (bert-base-lr_5e-05-trEpo5.0-MaxLen80-TrBN16-savStp_1000calGradiStp_2-warm_0)
What's the baseline for the BERT-BASE model? Any reference for 81? 
@szha @omerarshad   Thanks

**Comment by szha** [comment]
@mingtop thanks for sharing. The [original BERT paper](https://arxiv.org/pdf/1810.04805.pdf) reported 81.6 on dev set in table 4.
